src/BizHawk.Emulation.Cores/Consoles/Nintendo/Gameboy/GambatteLink.IEmulator.cs

```diff
@@ -23,7 +23,11 @@ public bool FrameAdvance(IController controller, bool render, bool rendersound =
 				{
 					for (int i = 0; i < _numCores; i++)
 					{
-						if (s.Contains($"P{i + 1} "))
+						if (s.Contains($"P{i + 1} -"))
+						{
+							_linkedConts[i].Set(s.Replace($"P{i + 1} - ", ""));
+						}
+						else if (s.Contains($"P{i + 1} "))

```

Comment:
This forced you to add a hack to `Bk2MnemonicLookup.cs`. What does it actually do? Can you implement it another way?

Reply:
This deals with the fact that an SGB core has 4 controllers, as shown in the last image in the original description. I don't have a better implementation in mind for now.

Reply:
Number them 1-4 and make the next console start from 5

Reply:
That wouldn't be intuitive on which core a controller corresponds to, especially when console modes are mixed.

Reply:
![image](https://user-images.githubusercontent.com/19356108/176623207-3a689cc2-6266-4c5a-b44b-f7909010479f.png)